### PR TITLE
add line number option for filelogreceiver

### PR DESCRIPTION
**Pull request by sfc-gh-jikim**
**Description:** Adding an option to include line numbers as a record attribute to the filelogreceiver.

**Testing:** Add unit tests

**Documentation:** Add documentation on filelogreceiver for the new file line number option

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Please make sure CLA is valid. We usually do not review PRs until then.

File: pkg/stanza/fileconsumer/config.go [hunk 1]
```diff
@@ -258,6 +258,10 @@ func (c Config) validate() error {
 		return fmt.Errorf("'include_file_owner_name' or 'include_file_owner_group_name' it's not supported for windows: %w", err)
 	}
 
+	if c.IncludeFileLineNumber && (c.SplitConfig.LineStartPattern != "" || c.SplitConfig.LineEndPattern != "") {
+		return fmt.Errorf("`multiline` cannot be used with `include_file_line_number`")
+	}
```
Comment (djaglowski) [inline pkg/stanza/fileconsumer/config.go:263]: Rather than excluding multiline files, could we report this attribute as "record number" instead?

Comment (sfc-gh-jikim) [reply pkg/stanza/fileconsumer/config.go]: Yup makes sense, fixed

File: pkg/stanza/fileconsumer/attrs/attrs.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ const (
 	LogFilePathResolved   = "log.file.path_resolved"
 	LogFileOwnerName      = "log.file.owner.name"
 	LogFileOwnerGroupName = "log.file.owner.group.name"
+	LogFileRecordNumber   = "log.file.Record_number"
```
Comment (bogdandrutu) [inline pkg/stanza/fileconsumer/attrs/attrs.go:20]: Why capital "R"?

**Review by bogdandrutu — APPROVED** [review]

File: pkg/stanza/fileconsumer/attrs/attrs.go (hunk 2 above)
Comment (sfc-gh-jikim) [reply pkg/stanza/fileconsumer/attrs/attrs.go]: Just a typo, thanks for catching it

Comment (sfc-gh-jikim) [reply pkg/stanza/fileconsumer/attrs/attrs.go]: fixed

File: pkg/stanza/fileconsumer/internal/reader/reader.go [hunk 3]
```diff
@@ -87,6 +90,12 @@ func (r *Reader) ReadToEnd(ctx context.Context) {
 			continue
 		}
 
+		_, fileRecordNumEnabled := r.FileAttributes[attrs.LogFileRecordNumber]
```
[the remark this answers is not in the document]
Comment (sfc-gh-jikim) [reply pkg/stanza/fileconsumer/internal/reader/reader.go]: I think I was just trying to include it from the attributes config to keep it consistent with the other file related attributes but that makes more sense. Fixed

File: pkg/stanza/fileconsumer/internal/reader/reader.go [hunk 4]
```diff
@@ -11,6 +11,8 @@ import (
 	"io"
 	"os"
 
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/attrs"
+
 	"go.opentelemetry.io/collector/component"
 	"go.uber.org/zap"
```
Comment (djaglowski) [inline pkg/stanza/fileconsumer/internal/reader/reader.go:17]:
```suggestion
	"go.opentelemetry.io/collector/component"
	"go.uber.org/zap"

	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/attrs"
```

File: pkg/stanza/fileconsumer/internal/reader/reader.go [hunk 5]
```diff
@@ -14,6 +14,8 @@ import (
...
 
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/attrs"
+
```
Comment (djaglowski) [inline pkg/stanza/fileconsumer/internal/reader/reader.go:18]:
```suggestion
```

File: pkg/stanza/fileconsumer/internal/reader/reader.go [hunk 6]
```diff
@@ -14,6 +14,8 @@ import (
...
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/decode"
```
Comment (djaglowski) [inline pkg/stanza/fileconsumer/internal/reader/reader.go:19]:
```suggestion
	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/decode"
	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/attrs"
```

**Review by djaglowski — APPROVED** [review]

**Pull request merged**